test/integration/targets/docker_network/tasks/tests/overlay.yml

```diff
@@ -10,47 +10,52 @@
 ## overlay #########################################################
 ####################################################################
 
-# Overlay networks require swarm initialization before they'll work
-- name: swarm
-  docker_swarm:
-    state: present
-    advertise_addr: 192.168.1.1
-
-- name: overlay
-  docker_network:
-    name: "{{ nname_1 }}"
-    driver: overlay
-    driver_options:
-      com.docker.network.driver.overlay.vxlanid_list: "257"
-  register: overlay_1
-
-- name: overlay (idempotency)
-  docker_network:
-    name: "{{ nname_1 }}"
-    driver: overlay
-    driver_options:
-      com.docker.network.driver.overlay.vxlanid_list: "257"
-  register: overlay_2
-
-- name: overlay (change)
-  docker_network:
-    name: "{{ nname_1 }}"
-    driver: bridge
-  register: overlay_3
-
-- name: cleanup network
-  docker_network:
-    name: "{{ nname_1 }}"
-    state: absent
-    force: yes
-
-- name: cleanup swarm
-  docker_swarm:
-    state: absent
-    force: yes
-
-- assert:
-    that:
-    - overlay_1 is changed
-    - overlay_2 is not changed
-    - overlay_3 is changed
+- block:
+  # Overlay networks require swarm initialization before they'll work
+  - name: swarm
```

Comment:
Since you're modifying all these tests anyway, and for the sake of consistency, could you replace the `advertise_addr` here with `"{{ansible_default_ipv4.address}}"`? I didn't know about this property when I wrote the test initially, and the default address could help make the tests more portable going forward.

Reply:
Good idea! I've updated it.